### PR TITLE
GCI-142: Refactored to not nest more than 3 if/for/while/switch/try statements

### PR DESCRIPTION
**Pull request by Bisht13**
## Description of what I changed
One of my changes from my PR #2780
Refactored to not nest more than 3 if/for/while/switch/try statements
## Issue I worked on
see https://issues.openmrs.org/browse/GCI-142
Task Instance https://codein.withgoogle.com/dashboard/task-instances/6187644498214912/

## Checklist: I completed these to help reviewers :)
- [x] My pull request only contains **ONE single commit**

- [x] My IDE is configured to follow the [**code style**]

- [x] I have **added tests** to cover my changes. 

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: api/src/main/java/org/openmrs/Obs.java [hunk 1]
```diff
@@ -980,7 +980,7 @@ public String getValueAsString(Locale locale) {
 				if (getValueCoded() == null) {
 					return "";
 				}
-				if (getValueDrug() != null) {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Obs.java]: Was this nested more than three times?

File: api/src/main/java/org/openmrs/Encounter.java [hunk 2]
```diff
@@ -172,14 +172,12 @@ public void setLocation(Location location) {
 		
 		if (obsParent.hasGroupMembers()) {
 			for (Obs child : obsParent.getGroupMembers()) {
-				if (!child.getVoided()) {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Encounter.java]: I now find this harder to read than before.

File: api/src/main/java/org/openmrs/Encounter.java [hunk 3]
```diff
@@ -172,14 +172,12 @@ public void setLocation(Location location) {
...
-					if (!child.isObsGrouping()) {
+				if (!child.getVoided() && !child.isObsGrouping()) {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Encounter.java]: I now find this harder to read than before.

Comment (Bisht13) [reply api/src/main/java/org/openmrs/Encounter.java]: But this causes it to be nested only thrice

Comment (dkayiwa) [reply api/src/main/java/org/openmrs/Encounter.java]: But do you agree that the code is now harder to understand?

Comment (Bisht13) [reply api/src/main/java/org/openmrs/Encounter.java]: Ya I do :pensive: how can I change it then without nesting more than 3 times

Comment (dkayiwa) [reply api/src/main/java/org/openmrs/Encounter.java]: I have no idea. But you could look for another place to apply this rule without making the code harder.

**Comment by Bisht13** [comment]
@dkayiwa  Hey I have corrected the mistakes. Please have a look 😃. I've submitted the task too in the GCI dashboard.

File: api/src/main/java/org/openmrs/Obs.java [hunk 4]
```diff
@@ -550,11 +550,9 @@ public void removeGroupMember(Obs member) {
 				parentObs = parentObs.getObsGroup();
 			}
 		} else if (this.getObsGroup() != null) {
-			for (Obs obsSibling : this.getObsGroup().getGroupMembers()) {
-				if (!obsSibling.isObsGrouping()) {
-					ret.add(obsSibling);
-				}
-			}
+                    this.getObsGroup().getGroupMembers().stream().filter((obsSibling) -> (!obsSibling.isObsGrouping())).forEachOrdered((obsSibling) -> {
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Obs.java]: Can you let the stream api build the collection instead of you manually adding the elements one by one?

Comment (dkayiwa) [reply api/src/main/java/org/openmrs/Obs.java]: Yes you can remove the nesting with streams and at the same time not have to manually add items to the collection.

**Comment by Bisht13** [comment]
@dkayiwa Hey i changed it. Can you have a look 😃

**Comment by dkayiwa** [comment]
@Bisht13 you have still not yet got me about streams. But since you have already made some research and good attempts, am approving your task, though not merging this pull request. 😊

**Comment by Bisht13** [comment]
I'll study more about them 😅

**Comment by dkayiwa** [comment]
@Bisht13 thanks for the humble and learning spirit! 👍

**Pull request closed without merging**